frontend/src/components/widgets/DataFrame/columns/ChartColumn.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  GridCell,
+  GridCellKind,
+  LoadingCell,
+} from "@glideapps/glide-data-grid"
+import { SparklineCellType } from "@glideapps/glide-data-grid-cells"
+
+import { isNullOrUndefined } from "src/lib/utils"
+
+import {
+  BaseColumn,
+  BaseColumnProps,
+  getErrorCell,
+  getEmptyCell,
+  ColumnCreator,
+  toSafeString,
+  toSafeArray,
+  mergeColumnParameters,
+  toSafeNumber,
+  formatNumber,
+} from "./utils"
+
+export interface ChartColumnParams {
+  // The type of the chart. Defaults to "line".
+  readonly type?: "line" | "bar"
```

Comment:
This probably could be an Enum. I don't think it needs to be in this PR but could be something as a TODO?

Reply:
Hmm, yes.. It's mostly an enum in a more condensed form since it only supports bar & line -> and will give a typing error if you provide a different value. But we might do this a bit differently if we migrate the column config to a proto structure (maybe something that I will do next quarter)